generators/dart/math.js

```diff
@@ -172,18 +172,58 @@ Blockly.Dart['math_constant'] = function(block) {
 Blockly.Dart['math_number_property'] = function(block) {
   // Check if a number is even, odd, prime, whole, positive, or negative
   // or if it is divisible by certain number. Returns true or false.
-  const number_to_check = Blockly.Dart.valueToCode(block, 'NUMBER_TO_CHECK',
-      Blockly.Dart.ORDER_MULTIPLICATIVE);
-  if (!number_to_check) {
-    return ['false', Blockly.Dart.ORDER_ATOMIC];
-  }
-  const dropdown_property = block.getFieldValue('PROPERTY');
-  let code;
-  if (dropdown_property === 'PRIME') {
+  const PROPERTIES = {
+
+    'EVEN': [' % 2 == 0', Blockly.Dart.ORDER_MULTIPLICATIVE,
+
+        Blockly.Dart.ORDER_EQUALITY],
+
+    'ODD': [' % 2 == 1', Blockly.Dart.ORDER_MULTIPLICATIVE,
+
+        Blockly.Dart.ORDER_EQUALITY],
+
+    'WHOLE': [' % 1 == 0', Blockly.Dart.ORDER_MULTIPLICATIVE,
+
+        Blockly.Dart.ORDER_EQUALITY],
+
+    'POSITIVE': [' > 0', Blockly.Dart.ORDER_RELATIONAL,
+
+        Blockly.Dart.ORDER_RELATIONAL],
+
+    'NEGATIVE': [' < 0', Blockly.Dart.ORDER_RELATIONAL,
+
+        Blockly.Dart.ORDER_RELATIONAL],
+
+    'DIVISIBLE_BY': [null, Blockly.Dart.ORDER_MULTIPLICATIVE,
+
+        Blockly.Dart.ORDER_EQUALITY],
+
+    'PRIME': [null, Blockly.Dart.ORDER_NONE,
+
+        Blockly.Dart.ORDER_UNARY_POSTFIX]
+
+  } 
+  const dropdownProperty = block.getFieldValue('PROPERTY');
+
+  const tuple = PROPERTIES[dropdownProperty];
+
+  const suffix = tuple[0];
+
+  const inputOrder = tuple[1];
+
+  const outputOrder = tuple[2];
+
+  const numberToCheck = Blockly.Dart.valueToCode(block, 'NUMBER_TO_CHECK',
+
+      inputOrder) || '0';
+
+      let code;
```

Comment:
Fix the indentation here so that it lines up with the `const` from the previous line.